sound/soc/sof/topology.c

```diff
@@ -1862,6 +1865,24 @@ static void sof_dai_set_format(struct snd_soc_tplg_hw_config *hw_config,
 	}
 }
 
+/* set config for all DAI's with name matching the link name */
+static void sof_set_dai_config(struct snd_sof_dev *sdev, u32 size,
+			       struct snd_soc_dai_link *link,
+			       struct sof_ipc_dai_config *config)
+{
+	struct snd_sof_dai *dai;
+
+	list_for_each_entry(dai, &sdev->dai_list, list) {
+		if (!dai->name)
+			continue;
+
+		if (strcmp(link->name, dai->name) == 0) {
+			dai->dai_config = kzalloc(size, GFP_KERNEL);
+			memcpy(dai->dai_config, config, size);
```

Comment:
@plbossart yes we do have a case where we do it multiple times. Esp for the ssp2 dai which is for headset playback and capture. I added it specifically when I made the PM updates because we were only update one previously and the other was failing during resume.

Reply:
@plbossart updated to use kmemdup